activerecord/lib/active_record/relation/where_clause.rb

```diff
@@ -26,15 +32,16 @@ def except(*columns)
       end
 
       def or(other)
-        if empty?
-          self
-        elsif other.empty?
-          other
-        else
-          WhereClause.new(
-            [ast.or(other.ast)],
-          )
-        end
+        left = self - other
+        common = self - left
+        right = other - common
+
+        return common if left.empty? || right.empty?
```

Comment:
I'd prefer to avoid postfix conditionals (and early returns for that matter) if possible. How about this instead?

```ruby
if left.empty? || right.empty?
  common
else
  or_clause = WhereClause.new(
    [left.ast.or(right.ast)],
  )
  common + or_clause
end
```